components/bl-stripe-payment-component/src/payment-components/payment-amount.js

```diff
@@ -1,14 +1,20 @@
-import { useRef } from 'react';
+import { useMemo } from 'react';
 
-import { validatePaymentAmount } from '../payment-form';
-
-export function PaymentAmount({ component, amount }) {
+export function PaymentAmount({ component, paymentAmount, setPaymentAmount }) {
   const { minAmount, fixedAmount, currency, amountDecimalPlaces } = component;
-  const amountRef = useRef(null);
 
-  const changeAmount = e => {
-    component.amount = validatePaymentAmount(e.target.value, amountDecimalPlaces);
-    amountRef.current.value = amount;
+  const amountPattern = useMemo(() => {
+    return new RegExp('^[0-9]*(\\.[0-9]{0,' + amountDecimalPlaces + '})?$');
+  }, [amountDecimalPlaces]);
```

Comment:
Looks like duplication with components/bl-stripe-payment-component/src/payment-form.js line 18-20
Only dollar sign at end is different

Could it be reusable hook that accept amount and options?